var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -266,8 +300,75 @@ def configure_args(self):
         if sys.platform == 'darwin':
             options.append('--with-build-config=bootstrap-debug')
 
+        # nvptx-none offloading for host compiler
+        if spec.satisfies('+nvptx'):
+            options.extend(['--enable-offload-targets=nvptx-none',
+                            '--with-cuda-driver-include={0}'.format(
+                                spec['cuda'].prefix.include),
+                            '--with-cuda-driver-lib={0}'.format(
+                                spec['cuda'].libs.directories[0]),
+                            '--disable-bootstrap',
+                            '--disable-multilib'])
+
         return options
 
+    # run configure/make/make(install) for the nvptx-none target
+    # before running the host compiler phases
+    @run_before('configure')
+    def nvptx_install(self):
+        spec = self.spec
+        prefix = self.prefix
+
+        if spec.satisfies('+nvptx'):
+
+            guess = Executable('./config.guess')
+            targetguess = guess(output=str).rstrip('\n')
```

Comment:
do you want to make this guess configurable? E.g. when compiling not on the actual compute node?

Reply:
ah ok, I might have not understood what `targetguess` stands for... device or host...

Reply:
config.guess returns the host, e.g. `x86_64-pc-linux-gnu`
We could potentially get this through the spack architecture spec property, but it would need a bunch of reformatting to get it into gcc's expected naming convention (I've experimented and it is very sensitive to needing to be exactly as expected). Calling guess seemed like the option to most often work - but yes, it assumes you are compiling on the same node type where you'll ultimately run

Reply:
Ok, but that triple looks quite portable and is not too likely to change between head/compile nodes and compute nodes.
Do you mind adding that triple example as a in-code comment for future readers?

Reply:
Fixed in commit 5252c10.